spec.bs

```diff
@@ -443,7 +445,9 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
         <b>The [=environment settings object/origin=]</b>
           1. Let |workletGlobalScope| be the [=global object=] of <var ignore=''>realmExecutionContext</var>'s Realm component.
           1. If |workletGlobalScope| is not {{SharedStorageWorkletGlobalScope}}, return |origin|.
-          1. Let |pendingAddedModules| be a [=list/clone=] of |worklet|'s [=added modules list=].
+          1. [=Assert=] that |worklet| is a {{SharedStorageWorklet}}.
+          1. If |worklet|'s [=SharedStorageWorklet/data origin=] is `"context-origin"`, return <var ignore=''>outsideSettings</var>'s [=environment settings object/origin=].
+          1. Else let |pendingAddedModules| be a [=list/clone=] of |worklet|'s [=added modules list=].
```

Comment:
`|pendingAddedModules|` should be defined outside the if/else scope

Reply:
Done.